src/test/java/uk/gov/companieshouse/overseasentitiesapi/converter/OverseasEntitySubmissionDaoConverterTest.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.companieshouse.overseasentitiesapi.converter;
+
+import org.bson.Document;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.mongodb.core.convert.MongoConverter;
+import uk.gov.companieshouse.overseasentitiesapi.model.SchemaVersion;
+import uk.gov.companieshouse.overseasentitiesapi.utils.TestUtils;
+
+import java.io.File;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class OverseasEntitySubmissionDaoConverterTest {
+
+    @InjectMocks
+    private OverseasEntitySubmissionDaoConverter overseasEntitySubmissionDaoConverter;
+
+    @Mock
+    private MongoConverter defaultMongoConverter;
+
+    @Mock
+    private TransformerFactory transformerFactory;
+
+    @Test
+    void testDocumentConversion() {
+        Document document = getDocument("overseas_entity_v_3_0.json");
+        overseasEntitySubmissionDaoConverter.convert(document);
+        verify(transformerFactory, times(1)).getTranformer(SchemaVersion.VERSION_3_0);
+        verify(defaultMongoConverter, times(1)).read(any(), eq(document));
```

Comment:
instead of any() in the verify, can you check that first param is of type OverseasEntitySubmissionDao.class

Reply:
Now using the class in the test.